core/src/main/java/org/elasticsearch/index/translog/TranslogReader.java

```diff
@@ -41,28 +41,55 @@
  * an immutable translog filereader
  */
 public class TranslogReader extends BaseTranslogReader implements Closeable {
+
     private static final byte LUCENE_CODEC_HEADER_BYTE = 0x3f;
     private static final byte UNVERSIONED_TRANSLOG_HEADER_BYTE = 0x00;
 
-
-    private final int totalOperations;
     protected final long length;
+    private final int totalOperations;
+    private final long minSeqNo;
+    private final long maxSeqNo;
     protected final AtomicBoolean closed = new AtomicBoolean(false);
 
     /**
      * Create a reader of translog file channel. The length parameter should be consistent with totalOperations and point
      * at the end of the last operation in this snapshot.
      */
-    public TranslogReader(long generation, FileChannel channel, Path path, long firstOperationOffset, long length, int totalOperations) {
+    public TranslogReader(
+        final long generation,
```

Comment:
I pushed 8786360f5f4f40fa31ab273464128abb34c440a4 and 48403f4baa62d1bd6223ea26a547ccb6b63ef6eb.